bson/src/test/unit/org/bson/codecs/pojo/ClassModelBuilderTest.java

```diff
@@ -107,10 +106,10 @@ public void testFieldsMappedClassTypes() {
         ClassModelBuilder<ConcreteCollectionsModel> builder =
                 ClassModel.builder(ConcreteCollectionsModel.class);
 
-        assertEquals(ArrayList.class, builder.getProperty("collection").getTypeData().getType());
-        assertEquals(ArrayList.class, builder.getProperty("list").getTypeData().getType());
+        assertEquals(Collection.class, builder.getProperty("collection").getTypeData().getType());
```

Comment:
We no longer need to alter the type data for interfaces to concrete implementations.